.changelog/12535.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+lib: add validation package + DNS label validation
```

Comment:
I don't think this should have a user-facing changelog entry, particularly given that the validation from this PR isn't being used yet